domain/consensus/processes/pruningmanager/pruningmanager.go

```diff
@@ -1133,3 +1127,37 @@ func (pm *pruningManager) isPruningPointInPruningDepth(stagingArea *model.Stagin
 
 	return blockGHOSTDAGData.BlueScore() >= pruningPointHeader.BlueScore()+pm.pruningDepth, nil
 }
+
+func (pm *pruningManager) TrustedBlockAssociatedGHOSTDAGDataBlockHashes(stagingArea *model.StagingArea, blockHash *externalapi.DomainHash) ([]*externalapi.DomainHash, error) {
+	blockHashes := make([]*externalapi.DomainHash, 0, pm.k)
+	current := blockHash
+	isTrustedData := false
+	for i := externalapi.KType(0); i < pm.k+1; i++ {
+		ghostdagData, err := pm.ghostdagDataStore.Get(pm.databaseContext, stagingArea, current, isTrustedData)
+		isNotFoundError := database.IsNotFoundError(err)
+		if !isNotFoundError && err != nil {
+			return nil, err
+		}
+		if isNotFoundError || ghostdagData.SelectedParent().Equal(model.VirtualGenesisBlockHash) {
+			isTrustedData = true
+			ghostdagData, err = pm.ghostdagDataStore.Get(pm.databaseContext, stagingArea, current, true)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		blockHashes = append(blockHashes, current)
+
+		if ghostdagData.SelectedParent().Equal(pm.genesisHash) {
+			break
+		}
+
+		if current.Equal(pm.genesisHash) {
+			break
+		}
+
+		current = ghostdagData.SelectedParent()
```

Comment:
Please make sure there aren't any scenarios where this can fail after the first prune.

Reply:
We don't prune headers yet, but when we do, we shouldn't delete the last K blocks from the selected chain of each pruning point anticone block.